php/php.dbgp/src/org/netbeans/modules/php/dbgp/ui/DbgpLineBreakpointCustomizerPanel.java

```diff
@@ -367,49 +398,77 @@ public boolean cancel() {
 
         @NbBundle.Messages({
             "CustomizerController.invalid.file=Existing file must be set.",
-            "CustomizerController.invalid.line=Valid line number must be set."
+            "CustomizerController.non.php.file=PHP source file is expected.",
+            "CustomizerController.invalid.line=Valid line number must be set.",
+            "# {0} - entered line number",
+            "# {1} - maximum line number in file",
+            "CustomizerController.too.big.line.number=The line number {0} is too big. Maximum line number is {1}."
         })
-        @Override
-        public boolean isValid() {
+        public void checkValid() {
             boolean isValid = true;
             // file
             String fileName = fileTextField.getText();
             if (fileName == null || fileName.trim().length() == 0) {
                 setErrorMessage(Bundle.CustomizerController_invalid_file());
-                return false;
+                setValid(false);
+                return;
             }
             File file = new File(fileName.trim());
             if (!file.exists()) {
                 setErrorMessage(Bundle.CustomizerController_invalid_file());
-                return false;
+                setValid(false);
+                return;
             }
             FileObject fileObject = FileUtil.toFileObject(file);
             if (fileObject == null) {
                 setErrorMessage(Bundle.CustomizerController_invalid_file());
-                return false;
+                setValid(false);
+                return;
+            }
+            else if (!FileUtils.isPhpFile(fileObject)) {
+                setErrorMessage(Bundle.CustomizerController_non_php_file());
+                setValid(false);
+                return;
             }
 
             // line number
             String lineNumberString = lineNumberTextField.getText();
             if (lineNumberString == null || lineNumberString.trim().length() == 0) {
                 setErrorMessage(Bundle.CustomizerController_invalid_line());
-                return false;
+                setValid(false);
+                return;
             }
             try {
                 int lineNumber = Integer.parseInt(lineNumberTextField.getText());
                 if (lineNumber <= 0) {
                     setErrorMessage(Bundle.CustomizerController_invalid_line());
-                    return false;
+                    setValid(false);
+                    return;
+                }
+                int maxLine = findNumLines(fileObject);
+                if (maxLine == 0) { // Not found
+                    maxLine = Integer.MAX_VALUE - 1; // Not to bother the user when we did not find it
+                }
+                if (lineNumber > maxLine + 1) {
```

Comment:
Why `+ 1`?

Reply:
Please leave the reason as a comment for someone in the future :)

Reply:
Done.